### PR TITLE
Hide remote actor functionality behind "remote" feature

### LINKED ISSUE #58: Add feature flag for distributed actors
Distributed actors in kameo requires using the `linkme` crate. However when adding CI to this project, the build failed on nightly due to `linkme`. Moving distributed actors behind a feature flag should solve this issue, and reduce dependencies for users who are only using actors locally.

### PR DESCRIPTION
**Pull request by shusvr**
Closes #58 

Not sure if I made all unnecessary dependencies optional, maybe you'll have some suggestions on which other dependencies can be disabled.

### CODE REVIEW

File: src/request/ask.rs [hunk 1]
```diff
@@ -35,16 +35,17 @@ where
...
     mailbox: &'a Mb,
     signal: Signal<A>,
-    rx: oneshot::Receiver<Result<BoxReply, BoxSendError>>,
+    rx: oneshot::Receiver<Result<BoxReply, error::BoxSendError>>,
 }
 
 /// A request to a remote actor.
 #[allow(missing_debug_implementations)]
+#[cfg(feature = "remote")]
 pub struct RemoteAskRequest<'a, A, M>
```
Comment (tqwewe) [inline src/request/ask.rs:44]: Seems like this type doesn't get marked as feature gated when documenting due to the missing `#[cfg_attr(feature = "doc", doc(cfg(feature = "remote")))]` annotation.

File: src/request.rs [hunk 2]
```diff
@@ -47,8 +47,14 @@ use futures::Future;
 mod ask;
 mod tell;
 
-pub use ask::{AskRequest, LocalAskRequest, RemoteAskRequest};
-pub use tell::{LocalTellRequest, RemoteTellRequest, TellRequest};
+#[cfg(feature = "remote")]
+pub use ask::RemoteAskRequest;
+
+#[cfg(feature = "remote")]
+pub use tell::RemoteTellRequest;
```
Comment (tqwewe) [inline src/request.rs:54]: These types don't get marked as feature gated due to the missing annotations.

**Review by tqwewe — CHANGES_REQUESTED** [review]
This is great! Thank you for taking the time to do this 😍 

I'm not sure that docs.rs will recognise enable a `feature = "doc"` flag. I belive these annotations should instead be `#[cfg_attr(docsrs, ...)]` instead based on the docs here:
<https://docs.rs/about/builds>.

However, it seems like we can just put:
```rust
#![cfg_attr(docsrs, feature(doc_auto_cfg))]
```
And none of the types need to be annotated with:
```rust
#[cfg_attr(feature = "doc", doc(cfg(feature = "remote")))]
```
This cleans things up a lot, making the feature gating in the docs automatically detected!

File: src/request/ask.rs (hunk 1 above)
Comment (tqwewe) [reply src/request/ask.rs:44]: Edit: if we go with the `doc_auto_cfg` suggestion, then we won't need this anymore

File: src/request.rs (hunk 2 above)
Comment (tqwewe) [reply src/request.rs:54]: Edit: if we go with the `doc_auto_cfg` suggestion, then we won't need this anymore

**Comment by shusvr** [comment]
Actually `feature = "doc"` would be used by docsrs since I used `all-features = true` in the manifest, but I do agree doc_auto_cfg is more convenient.

**Pull request merged**